src/solvers/dgsem_p4est/dg_3d_parallel.jl

```diff
@@ -7,8 +7,8 @@
 
 function rhs!(du, u, t,
               mesh::ParallelP4estMesh{3}, equations,
-              initial_condition, boundary_conditions, source_terms,
-              dg::DG, cache)
+              initial_condition, boundary_conditions, source_terms::Source,
+              dg::DG, cache) where {Source}
```

Comment:
Specializing on the source terms removes minor type instabilities and allocations.

Reply:
OK. I assume you added it to all corresponding places as well.